Client/Frontend/AuthenticationManager/PasscodeEntryViewController.swift

```diff
@@ -37,29 +37,32 @@ class PasscodeEntryViewController: BasePasscodeViewController {
             make.bottom.left.right.equalTo(self.view)
             make.top.equalTo(view.safeArea.top)
         }
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(willEnterForeground), name: NSNotification.Name.UIApplicationWillEnterForeground, object: nil)
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         passcodePane.codeInputView.delegate = self
 
-        // Don't show the keyboard or allow typing if we're locked out. Also display the error.
-        if authenticationInfo?.isLocked() ?? false {
-            displayLockoutError()
-            passcodePane.codeInputView.isUserInteractionEnabled = false
-        } else {
-            passcodePane.codeInputView.becomeFirstResponder()
-        }
+        disableInteractionIfPasscodeLocked()
     }
 
-    override func viewDidLayoutSubviews() {
-        super.viewDidLayoutSubviews()
-        
-        if authenticationInfo?.isLocked() ?? false {
+    // `viewWillAppear()` doesn't trigger when the view controller is already loaded
+    // and a user goes from background to foreground, `UIApplicationWillEnterForeground` handles this case.
+    @objc func willEnterForeground() {
+        disableInteractionIfPasscodeLocked()
+    }
+    
+    private func disableInteractionIfPasscodeLocked() {
+        if authenticationInfo?.isLocked() == true {
+            displayLockoutError()
```

Comment:
I know this can handle being called multiple times, just want to make sure there aren't weird UI stuff going on though.

Reply:
It's removed from superView each time it is called, should be safe.